probe-rs/targets/SAMD21.yaml

```diff
@@ -515,22 +608,44 @@ flash_algorithms:
       sectors:
         - size: 8192
           address: 0
+  atsamd21_128_eeprom:
+    name: atsamd21_128_eeprom
+    description: ATSAMD21 4kB Data EEPROM
+    default: true
+    instructions: QSEJBgpoUgcB1QQiCmAcShpJUWAbSUlECGAAIHBHACBwR0EIFkjBYRdJAYABfckH/NAAfYAHAdUBIHBHACBwRxC1EUwOSyo0HIAcfeQH/NDJHIkIiQAC4BDKCR8QwAAp+tEJSIAcGIAYfcAH/NAYfYAHAdUBIBC9ACAQvZ4ABAAAQABBBAAAABqlAAAAAAAAAAAAAA==
+    pc_init: 1
+    pc_uninit: 31
+    pc_program_page: 65
+    pc_erase_sector: 35
+    pc_erase_all: ~
+    data_section_offset: 140
+    flash_properties:
+      address_range:
+        start: 4194304
+        end: 4198400
+      page_size: 64
+      erased_byte_value: 255
+      program_page_timeout: 100
+      erase_sector_timeout: 1000
+      sectors:
+        - size: 256
+          address: 0
   atsamd21_256:
     name: atsamd21_256
     description: ATSAMD21 256kB Flash
     default: true
-    instructions: QSEJBgpoUgcB1QQiCmAtSitJUWAsSUlECGAAIHBHACBwRw8hiQMBQEIIDyBAAwJAELUkSMJhJUoCgAJ90gf80CJMASI/PJIDixgM4EoIwmEEgAJ90gf80AJ9kgcB1QEgEL3/MQExmULw0wAgEL3wtckciQgVS4kAEk3bHCuAK33bB/zQEU49PgApFtADRkApAdlAJAXgDEYD4IDKCR+AwyQfACz50S6AQDArfdsH/NArfZsH6NUBIPC9ACDwvQAAngAEAABAAEEEAAAAQaUAAAAAAAAAAAAA
+    instructions: QSEJBgpoUgcB1QQiCmAoSiZJUWAnSUlECGAAIHBHACBwRw8hiQMBQEIIDyBAAwJAELUfSMJhIEoCgAJ90gf80B1MASI/PJIDixgM4EoIwmEEgAJ90gf80AJ9kgcB1QEgEL3/MQExmULw0wAgEL0QtRFMD0vkHByAHH3kB/zQyRyJCIkAAuAQygkfEMAAKfrRCUg9OBiAGH3AB/zQGH2ABwHVASAQvQAgEL0AAJ4ABAAAQABBBAAAAEGlAAAAAAAAAAAAAA==
     pc_init: 1
     pc_uninit: 31
     pc_program_page: 111
     pc_erase_sector: 35
     pc_erase_all: ~
-    data_section_offset: 208
+    data_section_offset: 188
     flash_properties:
       address_range:
         start: 0
         end: 262144
-      page_size: 1024
+      page_size: 64
```

Comment:
This is probably going to resolve #507 (but I didn't have time to test that)

Reply:
Oh wow, I wonder if those bad page sizes were the culprit in many SAMD troubles we had :/ Maybe those are wrong for the 51 series too :/

Reply:
P.S. Would be great if you find the time to test #507 but no hurry ofc =)

Reply:
I've tested it on #507, but it does not work. 

```
Target /home/chief/code/led-backlighting-controller/target/thumbv6m-none-eabi/release/led-backlighting-controller
        WARN probe_rs::flashing::flasher > Chip erase was the selected method to erase the sectors but this chip does not support chip erases (yet).
        WARN probe_rs::flashing::flasher > A manual sector erase will be performed.
     Reading flash   ✔ [00:01:52] [###########################################################################################################################################]  21.35KB/ 21.35KB @     193B/s (eta 0s )
     Erasing sectors ✔ [00:00:00] [-------------------------------------------------------------------------------------------------------------------------------------------]       0B/ 32.00KB @       0B/s (eta 0s )
 Programming pages   ✔ [00:01:53] [-------------------------------------------------------------------------------------------------------------------------------------------]       0B/ 32.00KB @       0B/s (eta 0s )
       Error failed to flash /home/chief/code/led-backlighting-controller/target/thumbv6m-none-eabi/release/led-backlighting-controller
             
             Caused by:
                 0: Error while flashing
                 1: Failed to erase flash sector at address 0x00000000.
                 2: The execution of 'erase_sector' failed with code 1. This might indicate a problem with the flash algorithm.
```

I believe the flash algo for that chip is `atsamd21_128`, and the `flash_properties.address_rage` looks to be wrong I think?